plugins/find/dialogs/find.js

```diff
@@ -373,9 +375,27 @@
 					} else if ( !this._.state ) {
 						return KMP_NOMATCH;
 					} else {
-						this._.state = this._.overlap[this._.state];
+						this._.state = this._.overlap[ this._.state ];
 					}
 				}
+
+				/*
+					Whitespace comes in two forms: ' ' and &nbsp;
+					We should return true if one of them occurs.
+					There may be case like:
+
+						Search range: <p>ckeditor&nbsp test</p>
+						Search input: 'ckeditor  test'
+
+					In visual side there will be no difference. The difference is in UTF-16 codes. #4987
+				*/
+				function isWhiteSpace( character ) {
+					if ( character.charCodeAt( 0 ) === 160 || character.charCodeAt() === 32 ) {
```

Comment:
It's also worth checking if the issue is not occurring with other space characters, e.g. https://www.fileformat.info/info/unicode/category/Zs/list.htm
You can take a look at `isWordSeparator` function in the same file for possible other separators which may be incorrectly recognized.

Reply:
It seems that this piece of code is unnecessary because`isWordSeparator()` is working fine. Additionally, I add a few more unit tests to check each space separator.